respa_exchange/ews/session.py

```diff
@@ -64,13 +56,33 @@ def soap(self, request, timeout=10):
             "Accept": "text/xml",
             "Content-type": "text/xml; charset=%s" % self.encoding
         }
-        resp = self.post(self.url, data=body, headers=headers, auth=self.auth, timeout=timeout)
-        return self._process_soap_response(resp)
+        return dict(data=body, headers=headers, auth=self.auth)
+
+    def soap(self, request, timeout=10):
+        """
+        Send an EWSRequest by SOAP.
+
+        :type request: respa_exchange.base.EWSRequest
+        :param timeout: request timeout (see `requests` docs)
+        :type timeout: float|None|tuple[float, float]
+        :rtype: lxml.etree.Element
+        """
+        resp = self.post(self.url, timeout=timeout, **self._prepare_soap(request))
+        resp.raise_for_status()
+        return self._process_soap_response(resp.content)
+
+    def soap_stream(self, request, timeout=10):
+        """
+        Send an EWSRequest by SOAP and stream the response.
+        """
+        resp = self.post(self.url, timeout=timeout, stream=True, **self._prepare_soap(request))
+        for data in resp.iter_content(chunk_size=None):
```

Comment:
They always were during my testing. I'll experiment with lxml's incremental parsing and see if it's doable without too much of a hassle.